### ISSUE TITLE
question: why does gist encourage object properties to do the heavy lifting when it comes to expressing Roles? 

### ISSUE THREAD

**Issue by justin2004** [semanticarts/gist#695]
Hello!

In @uscholdm 's recent [Gist intro video](https://www.youtube.com/watch?v=YbaDZSuhm54) (quite helpful by the way) an example like the following was given:
```
[ a :LoanContract ;
  :hasBorrower :fred ;                                                                                                                  
  :hasLender :sally ]
```
So I would say that object properties are doing the work of conveying how Fred and Sally are involved in that loan.

An alternative representation is:
```
[ a :LoanContract ;                                                                                                                     
  :hasRole [ a :Borrower ;                                                                                                              
             :playedBy :fred ] ;                                                                                                        
  :hasRole [ a :Lender ;                                                                                                                
             :playedBy :sally ] ]                                                                                                       
``` 
And here classes are doing the work of conveying how Fred and Sally are involved in that loan.
Some nice things about this approach...

(1) You can more easily integrate with other ontologies where taxonomies are often expressed as small number of relationships between a large number of classes.
```
:Borrower owl:equivalentClass wd:Q157470 .                                                                                              
:Lender owl:equivalentClass wd:Q26726369 .
```
Wikidata has classes for borrowers, lenders, legal liability, etc. but it doesn't have object properties specifically for these so you can't easily integrate with Wikidata.

(2) You can qualify (talk about) Sally's role as a lender because you have a node there.
```
[ a :LoanContract ;                                                                                                                     
  :hasRole [ a :Borrower ;  
             :playedBy :fred ] ;
  :hasRole [ a :Lender ;                                                                                                                
             :playedBy :sally  ;
             :isGovernedBy wd:Q3112589 # Government of Florida
           ] ]
```
Perhaps when Sally is playing the role of lendor her actions are subject to Florida law but when Sally is playing the role of shortstop in her rec league's baseball tournament her actions are subject to the rec league's conditions. You can't make that distinct without an intermediate node for the role.

(3) There are also triplestore implementation concerns:
["If the number of distinct predicates in a graph becomes large, the default Neptune access strategy can become inefficient."](https://docs.aws.amazon.com/neptune/latest/userguide/feature-overview-data-model.html) I don't know how much this affects other implementations, however.

In general, I think it feels more natural to add classes/categories as needed rather than adding more object properties. In natural language if we do coin a new verb (~object property) it often _starts_ with a noun (~class or an individual): 

Google -> googles
Uber -> ubers
Instant Message -> IMs

But most nouns don't get a verb counterpart.

**Comment by justin2004** [comment]
At the moment I think `gist:hasParticipant` is really:
```
:hasRole/:playedBy
```

So I think we might need to add those 2 primitives to gist and then we can formally define gist:hasParticipant as a prop chain axiom.

**Comment by uscholdm** [comment]
> (1) You can more easily integrate with other ontologies where taxonomies are often expressed as small number of relationships between a large number of classes.

If the event that you need to link to external sources, you can do so when that need arises.   You can make a Borrower or Lender class in a separate mapping ontology.  We don't want to make complicated models because we might want to link to someone else's complicated model in the uncertain future.

**Comment by uscholdm** [comment]
> (2) You can qualify (talk about) Sally's role as a lender because you have a node there.

True, but again, you can extend the model if the need arises. No need to complicate thing just in case something arises in the future.  It is pretty rare for such things to arise in my experience.

**Comment by uscholdm** [comment]
> (3) There are also triplestore implementation concerns:

That is an interesting concern, but generally one does not want to make models that depend on or cater to certain implementation decisions that can change over time.

**Comment by rjyounes** [comment]
Add to a best patterns/design practices document. Only tangential to gist.
